lib/packetgen/header/http/response.rb

```diff
@@ -0,0 +1,116 @@
+# This file is part of PacketGen
+# See https://github.com/sdaubert/packetgen for more informations
+# Copyright (C) 2016 Sylvain Daubert <sylvain.daubert@laposte.net>
+# This program is published under MIT license.
+
+module PacketGen
+  module Header
+    module HTTP
+      # An HTTP/1.1 Respose packet consits of:
+      # * the version ({Types::String}).
+      # * the status code ({Types::String}).
+      # * the status message ({Types::String}).
+      # * associated http headers ({Types::HTTPHeaders}).
+      # * the actual http payload body ({Types::String}).
+      #
+      # == Create a HTTP Response header
+      #   # standalone
+      #   http_resp = PacketGen::Header::HTTP::Response.new
+      #   # in a packet
+      #   pkt = PacketGen.gen("IP").add("TCP").add("HTTP::Response")
+      #   # access to HTTP Response header
+      #   pkt.http_response # => PacketGen::Header::HTTP::Response
+      #
+      # == HTTP Response attributes
+      #   http_resp.version = "HTTP/1.1"
+      #   http_resp.status_code = "200"
+      #   http_resp.status_mesg = "OK"
+      #   http_resp.body = "this is a body"
+      #   http_resp.headers = "Host: tcpdump.org"     # string or
+      #   http_resp.headers = { host: "tcpdump.org" } # even a hash
+      #
+      # @author Kent 'picat' Gruber
+      class Response < Base
+        # @!attribute version 
+        #   @return [Types::String]
+        define_field :version,     Types::String, default: "HTTP/1.1"
+        # @!attribute status_code 
+        #   @return [Types::String]
+        define_field :status_code, Types::String
+        # @!attribute status_mesg 
+        #   @return [Types::String]
+        define_field :status_mesg, Types::String  
+        # @!attribute headers
+        #   associated http/1.1 headers
+        #   @return [Types::String]
+        define_field :headers, Types::HTTPHeaders
+        # @!attribute body 
+        #   @return [Types::HTTPHeaders]
+        define_field :body, Types::String
+        
+        # @param [Hash] options
+        # @option options [String] :version
+        # @option options [String] :status_code
+        # @option options [String] :status_mesg
+        # @option options [String] :body
+        # @option options [Hash]   :headers
+        def initialize(options={})
+          super(options)
+          self.headers ||= options[:headers]
+        end
+
+        # Give it a pretty name, the only way I know how.
+        # @return [String]
+        def protocol_name
+          "HTTP_Response"
+        end
```

Comment:
should return "HTTP::Response"

Reply:
It's the only way I know how the give it the `#http_response` associated with this packet type. 😟 

Otherwise it's just `#response` for some reason. 

I guess this could be patched elsewhere.

Reply:
OK. As said above, do not define #protocol_name